diff -Bbur nagios-4.0.20121116.orig/cgi/outages.c nagios-4.0.20121116/cgi/outages.c
--- nagios-4.0.20121116.orig/cgi/outages.c	2012-11-30 17:23:52.000000000 +0000
+++ nagios-4.0.20121116/cgi/outages.c	2012-11-30 17:24:51.000000000 +0000
@@ -424,9 +424,6 @@
 #ifdef USE_STATUSMAP
 		printf("<A HREF='%s?host=%s'><IMG SRC='%s%s' BORDER=0 ALT='View status map for this host and its children' TITLE='View status map for this host and its children'></A>\n", STATUSMAP_CGI, url_encode(temp_hostoutage->hst->name), url_images_path, STATUSMAP_ICON);
 #endif
-#ifdef USE_STATUSWRL
-		printf("<A HREF='%s?host=%s'><IMG SRC='%s%s' BORDER=0 ALT='View 3-D status map for this host and its children' TITLE='View 3-D status map for this host and its children'></A>\n", STATUSWORLD_CGI, url_encode(temp_hostoutage->hst->name), url_images_path, STATUSWORLD_ICON);
-#endif
 #ifdef USE_TRENDS
 		printf("<A HREF='%s?host=%s'><IMG SRC='%s%s' BORDER=0 ALT='View trends for this host' TITLE='View trends for this host'></A>\n", TRENDS_CGI, url_encode(temp_hostoutage->hst->name), url_images_path, TRENDS_ICON);
 #endif
