diff -ur nagiosgraph-0.7.original/insert.pl nagiosgraph-0.7/insert.pl
--- nagiosgraph-0.7.original/insert.pl	2005-10-26 15:42:57.000000000 +0100
+++ nagiosgraph-0.7/insert.pl	2006-03-27 16:13:33.000000000 +0100
@@ -9,7 +9,7 @@
 use RRDs;
 
 # Configuration
-my $configfile = '/usr/local/etc/nagiosgraph.conf';
+my $configfile = '/usr/local/nagios/etc/nagiosgraph.conf';
 
 # Main program - change nothing below
 
diff -ur nagiosgraph-0.7.original/nagiosgraph.conf nagiosgraph-0.7/nagiosgraph.conf
--- nagiosgraph-0.8.2.original/nagiosgraph.conf	2006-04-06 11:00:06.000000000 +0100
+++ nagiosgraph-0.8.2/nagiosgraph.conf	2006-07-21 11:48:21.400906096 +0100
@@ -13,22 +13,23 @@
 debug = 2
 
 # Location of debug log file
-logfile = /var/log/nagiosgraph.log
+logfile = /usr/local/nagios/var/log/nagiosgraph.log
 
 # Directory to store rrd database files
-rrddir =  /usr/local/nagios/nagiosgraph/rrd
+rrddir =  /usr/local/nagios/var/rrd
 
 # File containing regular expressions to identify service and perf data
-mapfile = /usr/local/nagios/nagiosgraph/map
+mapfile = /usr/local/nagios/etc/map
 
 # Color scheme for graphs. Choose a number between 1 and 8.
 colorscheme = 1
 
 # Heartbeat. In seconds, twice the size of servicecheck intervals
-heartbeat = 600
+# Altinity: set to 1 hour as a requirement for graphing
+heartbeat = 3600
 
-# Location of performance data log file. Comment out it not used.
-perflog = /var/spool/nagios/perfdata.log
+# Location of performance data log file. Comment out if not used.
+perflog = /usr/local/nagios/var/perfdata.log
 
 # Stylesheet - added to head of show.cgi. Comment out if not used
-stylesheet = /nagios/stylesheets/nagiosgraph.css
+#stylesheet = /nagios/stylesheets/admin.css
diff -ur nagiosgraph-0.7.original/show.cgi nagiosgraph-0.7/show.cgi
--- nagiosgraph-0.7.original/show.cgi	2005-10-24 07:29:34.000000000 +0100
+++ nagiosgraph-0.7/show.cgi	2006-03-27 16:13:48.000000000 +0100
@@ -10,7 +10,7 @@
 use CGI qw/:standard/;
 
 # Configuration
-my $configfile = '/usr/local/etc/nagiosgraph.conf';
+my $configfile = '/usr/local/nagios/etc/nagiosgraph.conf';
 
 # Main program - change nothing below
 
@@ -187,8 +187,8 @@
   #   Monthly =   5w =  3024000s
   #   Yearly  = 400d = 34560000s
   my @T=(['dai',118800], ['week',777600], ['month',3024000], ['year',34560000]);
-  print h1("Nagiosgraph");
-  print p("Performance data for ".strong("Host: ").tt($h).' &#183; '.strong("Service: ").tt($s));
+  print h1("Performance data");
+  print p(strong("Host: ").tt($h).' &#183; '.strong("Service: ").tt($s));
   for my $l ( @T ) {
     my($p,$t) = ($l->[0],$l->[1]);
     print start_div({-class => "graphs"});
