diff -Bbur nagios-4.0.20121116.orig/cgi/avail.c nagios-4.0.20121116/cgi/avail.c
--- nagios-4.0.20121116.orig/cgi/avail.c	2012-11-26 14:18:49.000000000 +0000
+++ nagios-4.0.20121116/cgi/avail.c	2012-11-26 16:16:17.000000000 +0000
@@ -1026,7 +1026,8 @@
 		printf("<select name='report_type'>\n");
 		printf("<option value=hostgroups>Hostgroup(s)\n");
 		printf("<option value=hosts>Host(s)\n");
-		printf("<option value=servicegroups>Servicegroup(s)\n");
+		/* Opsview patch: Remove servicegroups */
+		//printf("<option value=servicegroups>Servicegroup(s)\n");
 		printf("<option value=services>Service(s)\n");
 		printf("</select>\n");
 		printf("</td></tr>\n");
diff -Bbur nagios-4.0.20121116.orig/cgi/summary.c nagios-4.0.20121116/cgi/summary.c
--- nagios-4.0.20121116.orig/cgi/summary.c	2012-11-26 14:18:49.000000000 +0000
+++ nagios-4.0.20121116/cgi/summary.c	2012-11-26 16:16:17.000000000 +0000
@@ -583,6 +583,7 @@
 		printf("</select>\n");
 		printf("</td></tr>\n");
 
+		/* Opsview patch: Remove servicegroup option
 		printf("<tr><td class='reportSelectSubTitle' valign=center>Limit To Servicegroup:</td><td align=left valign=center class='reportSelectItem'>\n");
 		printf("<select name='servicegroup'>\n");
 		printf("<option value='all'>** ALL SERVICEGROUPS **\n");
@@ -592,6 +593,7 @@
 			}
 		printf("</select>\n");
 		printf("</td></tr>\n");
+		*/
 
 		printf("<tr><td class='reportSelectSubTitle' valign=center>Limit To Host:</td><td align=left valign=center class='reportSelectItem'>\n");
 		printf("<select name='host'>\n");
