diff -ur nagvis-1.4rc1.original/nagvis/includes/classes/GlobalMainCfg.php nagvis-1.4rc1/nagvis/includes/classes/GlobalMainCfg.php
--- nagvis-1.4rc1.original/nagvis/includes/classes/GlobalMainCfg.php	2008-11-27 21:48:35.000000000 +0000
+++ nagvis-1.4rc1/nagvis/includes/classes/GlobalMainCfg.php	2008-12-23 18:58:10.206891137 +0000
@@ -225,6 +225,10 @@
 					'editable' => 1,
 					'default' => '/nagios/nagvis',
 					'match' => MATCH_STRING_PATH),
+				'opsviewbase' => Array('must' => 0,
+					'editable' => 0,
+					'default' => '',
+					'match' => MATCH_STRING_PATH_EMPTY),
 				'htmlcgi' => Array('must' => 1,
 					'editable' => 1,
 					'default' => '/nagios/cgi-bin',
diff -ur nagvis-1.4rc1.original/nagvis/includes/defines/matches.php nagvis-1.4rc1/nagvis/includes/defines/matches.php
--- nagvis-1.4rc1.original/nagvis/includes/defines/matches.php	2008-11-09 13:16:36.000000000 +0000
+++ nagvis-1.4rc1/nagvis/includes/defines/matches.php	2008-12-23 18:57:46.477408137 +0000
@@ -32,6 +32,7 @@
 define('MATCH_STRING_NO_SPACE', '/^[0-9a-z\:\+\[\]\(\)\_\.\,\-\?\!\#\@\=\/\\\]+$/i');
 define('MATCH_STRING_NO_SPACE_EMPTY', '/^[0-9a-z\:\+\[\]\(\)\_\.\,\-\?\!\#\@\=\/\\\]*$/i');
 define('MATCH_STRING_PATH', '/^[0-9a-z\s\_\.\-\/\\\]+$/i');
+define('MATCH_STRING_PATH_EMPTY', '/^[0-9a-z\s\_\.\-\/\\\]*$/i');
 define('MATCH_STRING_URL', '/^[0-9a-z\s\:\+\[\]\(\)\=\%\?\&\_\.\-\#\@\=\/\\\]+$/i');
 define('MATCH_STRING_URL_EMPTY', '/^[0-9a-z\s\:\+\[\]\(\)\=\%\?\&\_\.\-\#\@\=\/\\\]*$/i');
 
