diff -Bbur ndoutils-2-0.orig/db/mysql.sql ndoutils-2-0/db/mysql.sql
--- ndoutils-2-0.orig/db/mysql.sql	2012-11-21 18:27:49.000000000 +0000
+++ ndoutils-2-0/db/mysql.sql	2012-11-21 18:45:48.000000000 +0000
@@ -33,13 +33,14 @@
   `object_id` int(11) NOT NULL default '0',
   `state` smallint(6) NOT NULL default '0',
   `author_name` varchar(64) character set latin1 NOT NULL default '',
-  `comment_data` varchar(255) character set latin1 NOT NULL default '',
+  `comment_data` text character set latin1 NOT NULL default '',
   `is_sticky` smallint(6) NOT NULL default '0',
   `persistent_comment` smallint(6) NOT NULL default '0',
   `notify_contacts` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`acknowledgement_id`),
-  UNIQUE_KEY `instance_id` (`entry_time`)
-) ENGINE=MyISAM COMMENT='Current and historical host and service acknowledgements';
+  UNIQUE_KEY `instance_id` (`entry_time`),
+  KEY `instance_object_id` (`instance_id`, `object_id`)
+) ENGINE=InnoDB COMMENT='Current and historical host and service acknowledgements';
 
 -- --------------------------------------------------------
 
@@ -55,7 +56,7 @@
   `command_line` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`command_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`object_id`,`config_type`)
-) ENGINE=MyISAM  COMMENT='Command definitions';
+) ENGINE=InnoDB  COMMENT='Command definitions';
 
 -- --------------------------------------------------------
 
@@ -74,7 +75,7 @@
   `comment_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `internal_comment_id` int(11) NOT NULL default '0',
   `author_name` varchar(64) character set latin1 NOT NULL default '',
-  `comment_data` varchar(255) character set latin1 NOT NULL default '',
+  `comment_data` text character set latin1 NOT NULL default '',
   `is_persistent` smallint(6) NOT NULL default '0',
   `comment_source` smallint(6) NOT NULL default '0',
   `expires` smallint(6) NOT NULL default '0',
@@ -83,7 +84,7 @@
   `deletion_time_usec` int(11) NOT NULL default '0',
   PRIMARY KEY  (`commenthistory_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`comment_time`,`internal_comment_id`)
-) ENGINE=MyISAM  COMMENT='Historical host and service comments';
+) ENGINE=InnoDB  COMMENT='Historical host and service comments';
 
 -- --------------------------------------------------------
 
@@ -102,14 +103,14 @@
   `comment_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `internal_comment_id` int(11) NOT NULL default '0',
   `author_name` varchar(64) character set latin1 NOT NULL default '',
-  `comment_data` varchar(255) character set latin1 NOT NULL default '',
+  `comment_data` text character set latin1 NOT NULL default '',
   `is_persistent` smallint(6) NOT NULL default '0',
   `comment_source` smallint(6) NOT NULL default '0',
   `expires` smallint(6) NOT NULL default '0',
   `expiration_time` datetime NOT NULL default '0000-00-00 00:00:00',
   PRIMARY KEY  (`comment_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`comment_time`,`internal_comment_id`)
-) ENGINE=MyISAM ;
+) ENGINE=InnoDB ;
 
 -- --------------------------------------------------------
 
@@ -123,8 +124,8 @@
   `configfile_type` smallint(6) NOT NULL default '0',
   `configfile_path` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`configfile_id`),
-  UNIQUE KEY `instance_id` (`instance_id`,`configfile_type`,`configfile_path`)
-) ENGINE=MyISAM  COMMENT='Configuration files';
+  UNIQUE KEY `instance_id` (`instance_id`,`configfile_id`)
+) ENGINE=InnoDB  COMMENT='Configuration files';
 
 -- --------------------------------------------------------
 
@@ -138,8 +139,9 @@
   `configfile_id` int(11) NOT NULL default '0',
   `varname` varchar(64) character set latin1 NOT NULL default '',
   `varvalue` varchar(255) character set latin1 NOT NULL default '',
-  PRIMARY KEY  (`configfilevariable_id`)
-) ENGINE=MyISAM  COMMENT='Configuration file variables';
+  PRIMARY KEY  (`configfilevariable_id`),
+  KEY `instance_id` (`instance_id`,`configfile_id`,`varname`)
+) ENGINE=InnoDB  COMMENT='Configuration file variables';
 
 -- --------------------------------------------------------
 
@@ -163,8 +165,9 @@
   `bytes_processed` int(11) NOT NULL default '0',
   `lines_processed` int(11) NOT NULL default '0',
   `entries_processed` int(11) NOT NULL default '0',
-  PRIMARY KEY  (`conninfo_id`)
-) ENGINE=MyISAM  COMMENT='NDO2DB daemon connection information';
+  PRIMARY KEY  (`conninfo_id`),
+  KEY `instance_id` (`instance_id`)
+) ENGINE=InnoDB  COMMENT='NDO2DB daemon connection information';
 
 -- --------------------------------------------------------
 
@@ -180,7 +183,7 @@
   `alias` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`contactgroup_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`contactgroup_object_id`)
-) ENGINE=MyISAM  COMMENT='Contactgroup definitions';
+) ENGINE=InnoDB  COMMENT='Contactgroup definitions';
 
 -- --------------------------------------------------------
 
@@ -195,7 +198,7 @@
   `contact_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`contactgroup_member_id`),
   UNIQUE KEY `instance_id` (`contactgroup_id`,`contact_object_id`)
-) ENGINE=MyISAM  COMMENT='Contactgroup members';
+) ENGINE=InnoDB  COMMENT='Contactgroup members';
 
 -- --------------------------------------------------------
 
@@ -215,7 +218,7 @@
   `command_args` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`contactnotificationmethod_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`contactnotification_id`,`start_time`,`start_time_usec`)
-) ENGINE=MyISAM  COMMENT='Historical record of contact notification methods';
+) ENGINE=InnoDB  COMMENT='Historical record of contact notification methods';
 
 -- --------------------------------------------------------
 
@@ -234,7 +237,7 @@
   `end_time_usec` int(11) NOT NULL default '0',
   PRIMARY KEY  (`contactnotification_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`contact_object_id`,`start_time`,`start_time_usec`)
-) ENGINE=MyISAM  COMMENT='Historical record of contact notifications';
+) ENGINE=InnoDB  COMMENT='Historical record of contact notifications';
 
 -- --------------------------------------------------------
 
@@ -268,7 +271,7 @@
   `notify_host_downtime` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`contact_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`contact_object_id`)
-) ENGINE=MyISAM  COMMENT='Contact definitions';
+) ENGINE=InnoDB  COMMENT='Contact definitions';
 
 -- --------------------------------------------------------
 
@@ -289,8 +292,9 @@
   `modified_host_attributes` int(11) NOT NULL default '0',
   `modified_service_attributes` int(11) NOT NULL default '0',
   PRIMARY KEY  (`contactstatus_id`),
-  UNIQUE KEY `contact_object_id` (`contact_object_id`)
-) ENGINE=MyISAM  COMMENT='Contact status';
+  UNIQUE KEY `contact_object_id` (`contact_object_id`),
+  KEY `instance_id` (`instance_id`)
+) ENGINE=InnoDB  COMMENT='Contact status';
 
 -- --------------------------------------------------------
 
@@ -306,7 +310,7 @@
   `address` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`contact_address_id`),
   UNIQUE KEY `contact_id` (`contact_id`,`address_number`)
-) ENGINE=MyISAM COMMENT='Contact addresses';
+) ENGINE=InnoDB COMMENT='Contact addresses';
 
 -- --------------------------------------------------------
 
@@ -323,7 +327,7 @@
   `command_args` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`contact_notificationcommand_id`),
   UNIQUE KEY `contact_id` (`contact_id`,`notification_type`,`command_object_id`,`command_args`)
-) ENGINE=MyISAM  COMMENT='Contact host and service notification commands';
+) ENGINE=InnoDB  COMMENT='Contact host and service notification commands';
 
 -- --------------------------------------------------------
 
@@ -342,7 +346,7 @@
   PRIMARY KEY  (`customvariable_id`),
   UNIQUE KEY `object_id_2` (`object_id`,`config_type`,`varname`),
   KEY `varname` (`varname`)
-) ENGINE=MyISAM COMMENT='Custom variables';
+) ENGINE=InnoDB COMMENT='Custom variables';
 
 -- --------------------------------------------------------
 
@@ -360,8 +364,9 @@
   `varvalue` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`customvariablestatus_id`),
   UNIQUE KEY `object_id_2` (`object_id`,`varname`),
-  KEY `varname` (`varname`)
-) ENGINE=MyISAM COMMENT='Custom variable status information';
+  KEY `varname` (`varname`),
+  KEY `instance_id` (`instance_id`)
+) ENGINE=InnoDB COMMENT='Custom variable status information';
 
 -- --------------------------------------------------------
 
@@ -372,7 +377,7 @@
 CREATE TABLE IF NOT EXISTS `nagios_dbversion` (
   `name` varchar(10) character set latin1 NOT NULL default '',
   `version` varchar(10) character set latin1 NOT NULL default ''
-) ENGINE=MyISAM;
+) ENGINE=InnoDB;
 
 -- --------------------------------------------------------
 
@@ -387,7 +392,7 @@
   `object_id` int(11) NOT NULL default '0',
   `entry_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `author_name` varchar(64) character set latin1 NOT NULL default '',
-  `comment_data` varchar(255) character set latin1 NOT NULL default '',
+  `comment_data` text character set latin1 NOT NULL default '',
   `internal_downtime_id` int(11) NOT NULL default '0',
   `triggered_by_id` int(11) NOT NULL default '0',
   `is_fixed` smallint(6) NOT NULL default '0',
@@ -402,7 +407,7 @@
   `was_cancelled` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`downtimehistory_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`object_id`,`entry_time`,`internal_downtime_id`)
-) ENGINE=MyISAM  COMMENT='Historical scheduled host and service downtime';
+) ENGINE=InnoDB  COMMENT='Historical scheduled host and service downtime';
 
 -- --------------------------------------------------------
 
@@ -428,11 +433,11 @@
   `early_timeout` smallint(6) NOT NULL default '0',
   `execution_time` double NOT NULL default '0',
   `return_code` smallint(6) NOT NULL default '0',
-  `output` varchar(255) character set latin1 NOT NULL default '',
+  `output` text character set latin1 NOT NULL default '',
   `long_output` TEXT NOT NULL default '',
   PRIMARY KEY  (`eventhandler_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`object_id`,`start_time`,`start_time_usec`)
-) ENGINE=MyISAM COMMENT='Historical host and service event handlers';
+) ENGINE=InnoDB COMMENT='Historical host and service event handlers';
 
 -- --------------------------------------------------------
 
@@ -446,9 +451,10 @@
   `entry_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `command_type` smallint(6) NOT NULL default '0',
   `command_name` varchar(128) character set latin1 NOT NULL default '',
-  `command_args` varchar(255) character set latin1 NOT NULL default '',
-  PRIMARY KEY  (`externalcommand_id`)
-) ENGINE=MyISAM  COMMENT='Historical record of processed external commands';
+  `command_args` text character set latin1 NOT NULL default '',
+  PRIMARY KEY  (`externalcommand_id`),
+  KEY `instance_id` (`instance_id`)
+) ENGINE=InnoDB  COMMENT='Historical record of processed external commands';
 
 -- --------------------------------------------------------
 
@@ -470,8 +476,9 @@
   `high_threshold` double NOT NULL default '0',
   `comment_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `internal_comment_id` int(11) NOT NULL default '0',
-  PRIMARY KEY  (`flappinghistory_id`)
-) ENGINE=MyISAM  COMMENT='Current and historical record of host and service flapping';
+  PRIMARY KEY  (`flappinghistory_id`),
+  KEY `instance_id` (`instance_id`, `object_id`)
+) ENGINE=InnoDB  COMMENT='Current and historical record of host and service flapping';
 
 -- --------------------------------------------------------
 
@@ -501,12 +508,12 @@
   `execution_time` double NOT NULL default '0',
   `latency` double NOT NULL default '0',
   `return_code` smallint(6) NOT NULL default '0',
-  `output` varchar(255) character set latin1 NOT NULL default '',
+  `output` text character set latin1 NOT NULL default '',
   `long_output` TEXT NOT NULL default '',
   `perfdata` TEXT character set latin1 NOT NULL default '',
   PRIMARY KEY  (`hostcheck_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`host_object_id`,`start_time`,`start_time_usec`)
-) ENGINE=MyISAM  COMMENT='Historical host checks';
+) ENGINE=InnoDB  COMMENT='Historical host checks';
 
 -- --------------------------------------------------------
 
@@ -528,7 +535,7 @@
   `fail_on_unreachable` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`hostdependency_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`host_object_id`,`dependent_host_object_id`,`dependency_type`,`inherits_parent`,`fail_on_up`,`fail_on_down`,`fail_on_unreachable`)
-) ENGINE=MyISAM COMMENT='Host dependency definitions';
+) ENGINE=InnoDB COMMENT='Host dependency definitions';
 
 -- --------------------------------------------------------
 
@@ -550,7 +557,7 @@
   `escalate_on_unreachable` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`hostescalation_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`host_object_id`,`timeperiod_object_id`,`first_notification`,`last_notification`)
-) ENGINE=MyISAM  COMMENT='Host escalation definitions';
+) ENGINE=InnoDB  COMMENT='Host escalation definitions';
 
 -- --------------------------------------------------------
 
@@ -565,7 +572,7 @@
   `contactgroup_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`hostescalation_contactgroup_id`),
   UNIQUE KEY `instance_id` (`hostescalation_id`,`contactgroup_object_id`)
-) ENGINE=MyISAM  COMMENT='Host escalation contact groups';
+) ENGINE=InnoDB  COMMENT='Host escalation contact groups';
 
 -- --------------------------------------------------------
 
@@ -580,7 +587,7 @@
   `contact_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`hostescalation_contact_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`hostescalation_id`,`contact_object_id`)
-) ENGINE=MyISAM ;
+) ENGINE=InnoDB ;
 
 -- --------------------------------------------------------
 
@@ -596,7 +603,7 @@
   `alias` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`hostgroup_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`hostgroup_object_id`)
-) ENGINE=MyISAM  COMMENT='Hostgroup definitions';
+) ENGINE=InnoDB  COMMENT='Hostgroup definitions';
 
 -- --------------------------------------------------------
 
@@ -611,7 +618,7 @@
   `host_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`hostgroup_member_id`),
   UNIQUE KEY `instance_id` (`hostgroup_id`,`host_object_id`)
-) ENGINE=MyISAM  COMMENT='Hostgroup members';
+) ENGINE=InnoDB  COMMENT='Hostgroup members';
 
 -- --------------------------------------------------------
 
@@ -681,7 +688,7 @@
   PRIMARY KEY  (`host_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`host_object_id`),
   KEY `host_object_id` (`host_object_id`)
-) ENGINE=MyISAM  COMMENT='Host definitions';
+) ENGINE=InnoDB  COMMENT='Host definitions';
 
 -- --------------------------------------------------------
 
@@ -694,7 +701,7 @@
   `instance_id` smallint(6) NOT NULL default '0',
   `host_object_id` int(11) NOT NULL default '0',
   `status_update_time` datetime NOT NULL default '0000-00-00 00:00:00',
-  `output` varchar(255) character set latin1 NOT NULL default '',
+  `output` text character set latin1 NOT NULL default '',
   `long_output` TEXT NOT NULL default '',
   `perfdata` TEXT character set latin1 NOT NULL default '',
   `current_state` smallint(6) NOT NULL default '0',
@@ -756,7 +763,7 @@
   KEY `latency` (`latency`),
   KEY `execution_time` (`execution_time`),
   KEY `scheduled_downtime_depth` (`scheduled_downtime_depth`)
-) ENGINE=MyISAM  COMMENT='Current host status information';
+) ENGINE=InnoDB  COMMENT='Current host status information';
 
 -- --------------------------------------------------------
 
@@ -771,7 +778,7 @@
   `contactgroup_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`host_contactgroup_id`),
   UNIQUE KEY `instance_id` (`host_id`,`contactgroup_object_id`)
-) ENGINE=MyISAM  COMMENT='Host contact groups';
+) ENGINE=InnoDB  COMMENT='Host contact groups';
 
 -- --------------------------------------------------------
 
@@ -786,7 +793,7 @@
   `contact_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`host_contact_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`host_id`,`contact_object_id`)
-) ENGINE=MyISAM ;
+) ENGINE=InnoDB ;
 
 -- --------------------------------------------------------
 
@@ -801,7 +808,7 @@
   `parent_host_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`host_parenthost_id`),
   UNIQUE KEY `instance_id` (`host_id`,`parent_host_object_id`)
-) ENGINE=MyISAM  COMMENT='Parent hosts';
+) ENGINE=InnoDB  COMMENT='Parent hosts';
 
 -- --------------------------------------------------------
 
@@ -814,7 +821,7 @@
   `instance_name` varchar(64) character set latin1 NOT NULL default '',
   `instance_description` varchar(128) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`instance_id`)
-) ENGINE=MyISAM  COMMENT='Location names of various Nagios installations';
+) ENGINE=InnoDB  COMMENT='Location names of various Nagios installations';
 
 -- --------------------------------------------------------
 
@@ -829,12 +836,14 @@
   `entry_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `entry_time_usec` int(11) NOT NULL default '0',
   `logentry_type` int(11) NOT NULL default '0',
-  `logentry_data` varchar(255) character set latin1 NOT NULL default '',
+  `logentry_data` text character set latin1 NOT NULL default '',
   `realtime_data` smallint(6) NOT NULL default '0',
   `inferred_data_extracted` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`logentry_id`),
-  UNIQUE KEY `instance_id` (`logentry_time`)
-) ENGINE=MyISAM COMMENT='Historical record of log entries';
+  UNIQUE KEY `instance_id` (`logentry_time`),
+  KEY `instance_id2` (`instance_id`),
+  KEY `entry_time` (`entry_time`)
+) ENGINE=InnoDB COMMENT='Historical record of log entries';
 
 -- --------------------------------------------------------
 
@@ -853,13 +862,13 @@
   `end_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `end_time_usec` int(11) NOT NULL default '0',
   `state` smallint(6) NOT NULL default '0',
-  `output` varchar(255) character set latin1 NOT NULL default '',
+  `output` text character set latin1 NOT NULL default '',
   `long_output` TEXT NOT NULL default '',
   `escalated` smallint(6) NOT NULL default '0',
   `contacts_notified` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`notification_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`object_id`,`start_time`,`start_time_usec`)
-) ENGINE=MyISAM  COMMENT='Historical record of host and service notifications';
+) ENGINE=InnoDB  COMMENT='Historical record of host and service notifications';
 
 -- --------------------------------------------------------
 
@@ -875,8 +884,9 @@
   `name2` varchar(128) character set latin1 default NULL,
   `is_active` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`object_id`),
-  KEY `objecttype_id` (`objecttype_id`,`name1`,`name2`)
-) ENGINE=MyISAM  COMMENT='Current and historical objects of all kinds';
+  KEY `objecttype_id` (`objecttype_id`,`name1`,`name2`),
+  KEY `instance_id` (`instance_id`)
+) ENGINE=InnoDB  COMMENT='Current and historical objects of all kinds';
 
 -- --------------------------------------------------------
 
@@ -894,8 +904,10 @@
   `program_name` varchar(16) character set latin1 NOT NULL default '',
   `program_version` varchar(20) character set latin1 NOT NULL default '',
   `program_date` varchar(10) character set latin1 NOT NULL default '',
-  PRIMARY KEY  (`processevent_id`)
-) ENGINE=MyISAM  COMMENT='Historical Nagios process events';
+  PRIMARY KEY  (`processevent_id`),
+  KEY `instance_id` (`instance_id`, `event_type`),
+  KEY `event_time` (`event_time`, `event_time_usec`)
+) ENGINE=InnoDB  COMMENT='Historical Nagios process events';
 
 -- --------------------------------------------------------
 
@@ -931,7 +943,7 @@
   `global_service_event_handler` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`programstatus_id`),
   UNIQUE KEY `instance_id` (`instance_id`)
-) ENGINE=MyISAM  COMMENT='Current program status information';
+) ENGINE=InnoDB  COMMENT='Current program status information';
 
 -- --------------------------------------------------------
 
@@ -946,7 +958,7 @@
   `varvalue` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`runtimevariable_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`varname`)
-) ENGINE=MyISAM  COMMENT='Runtime variables from the Nagios daemon';
+) ENGINE=InnoDB  COMMENT='Runtime variables from the Nagios daemon';
 
 -- --------------------------------------------------------
 
@@ -961,7 +973,7 @@
   `object_id` int(11) NOT NULL default '0',
   `entry_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `author_name` varchar(64) character set latin1 NOT NULL default '',
-  `comment_data` varchar(255) character set latin1 NOT NULL default '',
+  `comment_data` text character set latin1 NOT NULL default '',
   `internal_downtime_id` int(11) NOT NULL default '0',
   `triggered_by_id` int(11) NOT NULL default '0',
   `is_fixed` smallint(6) NOT NULL default '0',
@@ -973,7 +985,7 @@
   `actual_start_time_usec` int(11) NOT NULL default '0',
   PRIMARY KEY  (`scheduleddowntime_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`object_id`,`entry_time`,`internal_downtime_id`)
-) ENGINE=MyISAM COMMENT='Current scheduled host and service downtime';
+) ENGINE=InnoDB COMMENT='Current scheduled host and service downtime';
 
 -- --------------------------------------------------------
 
@@ -1002,14 +1014,14 @@
   `execution_time` double NOT NULL default '0',
   `latency` double NOT NULL default '0',
   `return_code` smallint(6) NOT NULL default '0',
-  `output` varchar(255) character set latin1 NOT NULL default '',
+  `output` text character set latin1 NOT NULL default '',
   `long_output` TEXT NOT NULL default '',
   `perfdata` TEXT character set latin1 NOT NULL default '',
   PRIMARY KEY  (`servicecheck_id`),
   KEY `instance_id` (`instance_id`),
   KEY `service_object_id` (`service_object_id`),
   KEY `start_time` (`start_time`)
-) ENGINE=MyISAM  COMMENT='Historical service checks';
+) ENGINE=InnoDB  COMMENT='Historical service checks';
 
 -- --------------------------------------------------------
 
@@ -1032,7 +1044,7 @@
   `fail_on_critical` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`servicedependency_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`service_object_id`,`dependent_service_object_id`,`dependency_type`,`inherits_parent`,`fail_on_ok`,`fail_on_warning`,`fail_on_unknown`,`fail_on_critical`)
-) ENGINE=MyISAM COMMENT='Service dependency definitions';
+) ENGINE=InnoDB COMMENT='Service dependency definitions';
 
 -- --------------------------------------------------------
 
@@ -1055,7 +1067,7 @@
   `escalate_on_critical` smallint(6) NOT NULL default '0',
   PRIMARY KEY  (`serviceescalation_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`service_object_id`,`timeperiod_object_id`,`first_notification`,`last_notification`)
-) ENGINE=MyISAM  COMMENT='Service escalation definitions';
+) ENGINE=InnoDB  COMMENT='Service escalation definitions';
 
 -- --------------------------------------------------------
 
@@ -1070,7 +1082,7 @@
   `contactgroup_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`serviceescalation_contactgroup_id`),
   UNIQUE KEY `instance_id` (`serviceescalation_id`,`contactgroup_object_id`)
-) ENGINE=MyISAM  COMMENT='Service escalation contact groups';
+) ENGINE=InnoDB  COMMENT='Service escalation contact groups';
 
 -- --------------------------------------------------------
 
@@ -1085,7 +1097,7 @@
   `contact_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`serviceescalation_contact_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`serviceescalation_id`,`contact_object_id`)
-) ENGINE=MyISAM ;
+) ENGINE=InnoDB ;
 
 -- --------------------------------------------------------
 
@@ -1101,7 +1113,7 @@
   `alias` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`servicegroup_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`servicegroup_object_id`)
-) ENGINE=MyISAM  COMMENT='Servicegroup definitions';
+) ENGINE=InnoDB  COMMENT='Servicegroup definitions';
 
 -- --------------------------------------------------------
 
@@ -1116,7 +1128,7 @@
   `service_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`servicegroup_member_id`),
   UNIQUE KEY `instance_id` (`servicegroup_id`,`service_object_id`)
-) ENGINE=MyISAM  COMMENT='Servicegroup members';
+) ENGINE=InnoDB  COMMENT='Servicegroup members';
 
 -- --------------------------------------------------------
 
@@ -1180,7 +1192,7 @@
   PRIMARY KEY  (`service_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`service_object_id`),
   KEY `service_object_id` (`service_object_id`)
-) ENGINE=MyISAM  COMMENT='Service definitions';
+) ENGINE=InnoDB  COMMENT='Service definitions';
 
 -- --------------------------------------------------------
 
@@ -1193,7 +1205,7 @@
   `instance_id` smallint(6) NOT NULL default '0',
   `service_object_id` int(11) NOT NULL default '0',
   `status_update_time` datetime NOT NULL default '0000-00-00 00:00:00',
-  `output` varchar(255) character set latin1 NOT NULL default '',
+  `output` text character set latin1 NOT NULL default '',
   `long_output` TEXT NOT NULL default '',
   `perfdata` TEXT character set latin1 NOT NULL default '',
   `current_state` smallint(6) NOT NULL default '0',
@@ -1255,8 +1267,9 @@
   KEY `percent_state_change` (`percent_state_change`),
   KEY `latency` (`latency`),
   KEY `execution_time` (`execution_time`),
-  KEY `scheduled_downtime_depth` (`scheduled_downtime_depth`)
-) ENGINE=MyISAM  COMMENT='Current service status information';
+  KEY `scheduled_downtime_depth` (`scheduled_downtime_depth`),
+  KEY `instance_id2` (`instance_id`, `service_object_id` )
+) ENGINE=InnoDB  COMMENT='Current service status information';
 
 -- --------------------------------------------------------
 
@@ -1271,7 +1284,7 @@
   `contactgroup_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`service_contactgroup_id`),
   UNIQUE KEY `instance_id` (`service_id`,`contactgroup_object_id`)
-) ENGINE=MyISAM  COMMENT='Service contact groups';
+) ENGINE=InnoDB  COMMENT='Service contact groups';
 
 -- --------------------------------------------------------
 
@@ -1286,7 +1299,7 @@
   `contact_object_id` int(11) NOT NULL default '0',
   PRIMARY KEY  (`service_contact_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`service_id`,`contact_object_id`)
-) ENGINE=MyISAM ;
+) ENGINE=InnoDB ;
 
 -- --------------------------------------------------------
 
@@ -1307,10 +1320,11 @@
   `max_check_attempts` smallint(6) NOT NULL default '0',
   `last_state` smallint(6) NOT NULL default '-1',
   `last_hard_state` smallint(6) NOT NULL default '-1',
-  `output` varchar(255) character set latin1 NOT NULL default '',
+  `output` text character set latin1 NOT NULL default '',
   `long_output` TEXT NOT NULL default '',
-  PRIMARY KEY  (`statehistory_id`)
-) ENGINE=MyISAM COMMENT='Historical host and service state changes';
+  PRIMARY KEY  (`statehistory_id`),
+  KEY `instance_id` (`instance_id`, `object_id`)
+) ENGINE=InnoDB COMMENT='Historical host and service state changes';
 
 -- --------------------------------------------------------
 
@@ -1330,12 +1344,12 @@
   `early_timeout` smallint(6) NOT NULL default '0',
   `execution_time` double NOT NULL default '0',
   `return_code` smallint(6) NOT NULL default '0',
-  `output` varchar(255) character set latin1 NOT NULL default '',
+  `output` text character set latin1 NOT NULL default '',
   `long_output` TEXT NOT NULL default '',
   PRIMARY KEY  (`systemcommand_id`),
   KEY `instance_id` (`instance_id`),
   KEY `start_time` (`start_time`)
-) ENGINE=MyISAM  COMMENT='Historical system commands that are executed';
+) ENGINE=InnoDB  COMMENT='Historical system commands that are executed';
 
 -- --------------------------------------------------------
 
@@ -1357,7 +1371,7 @@
   KEY `event_type` (`event_type`),
   KEY `scheduled_time` (`scheduled_time`),
   KEY `object_id` (`object_id`)
-) ENGINE=MyISAM  COMMENT='Current Nagios event queue';
+) ENGINE=InnoDB  COMMENT='Current Nagios event queue';
 
 -- --------------------------------------------------------
 
@@ -1383,7 +1397,7 @@
   KEY `event_type` (`event_type`),
   KEY `scheduled_time` (`scheduled_time`),
   KEY `object_id` (`object_id`)
-) ENGINE=MyISAM  COMMENT='Historical events from the Nagios event queue';
+) ENGINE=InnoDB  COMMENT='Historical events from the Nagios event queue';
 
 -- --------------------------------------------------------
 
@@ -1399,7 +1413,7 @@
   `alias` varchar(255) character set latin1 NOT NULL default '',
   PRIMARY KEY  (`timeperiod_id`),
   UNIQUE KEY `instance_id` (`instance_id`,`config_type`,`timeperiod_object_id`)
-) ENGINE=MyISAM  COMMENT='Timeperiod definitions';
+) ENGINE=InnoDB  COMMENT='Timeperiod definitions';
 
 -- --------------------------------------------------------
 
@@ -1416,4 +1430,4 @@
   `end_sec` int(11) NOT NULL default '0',
   PRIMARY KEY  (`timeperiod_timerange_id`),
   UNIQUE KEY `instance_id` (`timeperiod_id`,`day`,`start_sec`,`end_sec`)
-) ENGINE=MyISAM  COMMENT='Timeperiod definitions';
+) ENGINE=InnoDB  COMMENT='Timeperiod definitions';
