diff -ur ndoutils-1.4b7.orig/src/dbhandlers.c ndoutils-1.4b7/src/dbhandlers.c
--- ndoutils-1.4b7.orig/src/dbhandlers.c	2009-04-21 14:04:07.000000000 +0000
+++ ndoutils-1.4b7/src/dbhandlers.c	2009-04-22 09:44:48.000000000 +0000
@@ -686,6 +686,10 @@
 		ndo2db_db_clear_table(idi,ndo2db_db_tablenames[NDO2DB_DBTABLE_HOSTCONTACTS]);
 		ndo2db_db_clear_table(idi,ndo2db_db_tablenames[NDO2DB_DBTABLE_SERVICES]);
 		ndo2db_db_clear_table(idi,ndo2db_db_tablenames[NDO2DB_DBTABLE_SERVICECONTACTS]);
+		ndo2db_db_clear_table(idi,ndo2db_db_tablenames[NDO2DB_DBTABLE_SERVICECONTACTGROUPS]);
+		ndo2db_db_clear_table(idi,ndo2db_db_tablenames[NDO2DB_DBTABLE_HOSTCONTACTGROUPS]);
+		ndo2db_db_clear_table(idi,ndo2db_db_tablenames[NDO2DB_DBTABLE_HOSTESCALATIONCONTACTGROUPS]);
+		ndo2db_db_clear_table(idi,ndo2db_db_tablenames[NDO2DB_DBTABLE_SERVICEESCALATIONCONTACTGROUPS]);
 
 		/* flag all objects as being inactive */
 		ndo2db_set_all_objects_as_inactive(idi);
