diff -Bbur ndoutils-2-0.orig/db/mysql.sql ndoutils-2-0/db/mysql.sql
--- ndoutils-2-0.orig/db/mysql.sql	2012-11-21 17:39:15.000000000 +0000
+++ ndoutils-2-0/db/mysql.sql	2012-11-21 17:39:27.000000000 +0000
@@ -847,6 +847,7 @@
   `instance_id` smallint(6) NOT NULL default '0',
   `notification_type` smallint(6) NOT NULL default '0',
   `notification_reason` smallint(6) NOT NULL default '0',
+  `notification_number` smallint(6) NOT NULL default '0',
   `object_id` int(11) NOT NULL default '0',
   `start_time` datetime NOT NULL default '0000-00-00 00:00:00',
   `start_time_usec` int(11) NOT NULL default '0',
diff -Bbur ndoutils-2-0.orig/include/nagios-4x/nebstructs.h ndoutils-2-0/include/nagios-4x/nebstructs.h
--- ndoutils-2-0.orig/include/nagios-4x/nebstructs.h	2012-11-21 17:39:15.000000000 +0000
+++ ndoutils-2-0/include/nagios-4x/nebstructs.h	2012-11-21 17:40:25.000000000 +0000
@@ -326,6 +326,7 @@
 	char            *ack_data;
 	int             escalated;
 	int             contacts_notified;
+	int             notification_number;
 
 	void            *object_ptr;
 	} nebstruct_notification_data;
diff -Bbur ndoutils-2-0.orig/src/dbhandlers.c ndoutils-2-0/src/dbhandlers.c
--- ndoutils-2-0.orig/src/dbhandlers.c	2012-11-21 17:39:15.000000000 +0000
+++ ndoutils-2-0/src/dbhandlers.c	2012-11-21 17:39:27.000000000 +0000
@@ -1184,6 +1184,7 @@
 	struct timeval tstamp;
 	int notification_type=0;
 	int notification_reason=0;
+	int notification_number=0;
 	unsigned long object_id=0L;
 	struct timeval start_time;
 	struct timeval end_time;
@@ -1205,6 +1206,7 @@
 
 	/* convert vars */
 	result=ndo2db_convert_string_to_int(idi->buffered_input[NDO_DATA_NOTIFICATIONTYPE],&notification_type);
+	result=ndo2db_convert_string_to_int(idi->buffered_input[NDO_DATA_CURRENTNOTIFICATIONNUMBER],&notification_number);
 	result=ndo2db_convert_string_to_int(idi->buffered_input[NDO_DATA_NOTIFICATIONREASON],&notification_reason);
 	result=ndo2db_convert_string_to_int(idi->buffered_input[NDO_DATA_STATE],&state);
 	result=ndo2db_convert_string_to_int(idi->buffered_input[NDO_DATA_ESCALATED],&escalated);
@@ -1226,10 +1228,11 @@
 		result=ndo2db_get_object_id_with_insert(idi,NDO2DB_OBJECTTYPE_HOST,idi->buffered_input[NDO_DATA_HOST],NULL,&object_id);
 
 	/* save entry to db */
-	if(asprintf(&buf,"instance_id='%lu', notification_type='%d', notification_reason='%d', start_time=%s, start_time_usec='%lu', end_time=%s, end_time_usec='%lu', object_id='%lu', state='%d', output='%s', long_output='%s', escalated='%d', contacts_notified='%d'"
+	if(asprintf(&buf,"instance_id='%lu', notification_type='%d', notification_reason='%d', notification_number='%d', start_time=%s, start_time_usec='%lu', end_time=%s, end_time_usec='%lu', object_id='%lu', state='%d', output='%s', long_output='%s', escalated='%d', contacts_notified='%d'"
 		    ,idi->dbinfo.instance_id
 		    ,notification_type
 		    ,notification_reason
+		    ,notification_number
 		    ,ts[0]
 		    ,start_time.tv_usec
 		    ,ts[1]
diff -Bbur ndoutils-2-0.orig/src/ndomod.c ndoutils-2-0/src/ndomod.c
--- ndoutils-2-0.orig/src/ndomod.c	2012-11-21 17:39:15.000000000 +0000
+++ ndoutils-2-0/src/ndomod.c	2012-11-21 17:39:27.000000000 +0000
@@ -1795,7 +1795,7 @@
 		es[5]=ndo_escape_buffer(notdata->ack_data);
 
 		snprintf(temp_buffer,sizeof(temp_buffer)-1
-			 ,"\n%d:\n%d=%d\n%d=%d\n%d=%d\n%d=%ld.%ld\n%d=%d\n%d=%ld.%ld\n%d=%ld.%ld\n%d=%s\n%d=%s\n%d=%d\n%d=%d\n%d=%s\n%d=%s\n%d=%s\n%d=%s\n%d=%d\n%d=%d\n%d\n\n"
+			 ,"\n%d:\n%d=%d\n%d=%d\n%d=%d\n%d=%ld.%ld\n%d=%d\n%d=%ld.%ld\n%d=%ld.%ld\n%d=%s\n%d=%s\n%d=%d\n%d=%d\n%d=%d\n%d=%s\n%d=%s\n%d=%s\n%d=%s\n%d=%d\n%d=%d\n%d\n\n"
 			 ,NDO_API_NOTIFICATIONDATA
 			 ,NDO_DATA_TYPE
 			 ,notdata->type
@@ -1820,6 +1820,8 @@
 			 ,(es[1]==NULL)?"":es[1]
 			 ,NDO_DATA_NOTIFICATIONREASON
 			 ,notdata->reason_type
+			 ,NDO_DATA_CURRENTNOTIFICATIONNUMBER
+			 ,notdata->notification_number
 			 ,NDO_DATA_STATE
 			 ,notdata->state
 			 ,NDO_DATA_OUTPUT
