diff -ur nmis-4.2.13/cgi-bin/nmiscgi.pl nmis-4.2.13.with_opsview_link/cgi-bin/nmiscgi.pl
--- nmis-4.2.13/cgi-bin/nmiscgi.pl	2008-11-14 18:18:50.000000000 +0800
+++ nmis-4.2.13.with_opsview_link/cgi-bin/nmiscgi.pl	2008-11-14 18:20:35.000000000 +0800
@@ -3808,6 +3808,9 @@
 	# get localtime
 	$time = &get_localtime;
 	print "<!-- nmisMenuSmall begin -->\n";
+	# Use parameters from instance.cfg
+	use Opsview::Slave::Config;
+	my $opsview_base = Opsview::Slave::Config->override_base_prefix || "/";
 	print <<EO_HTML;
 <tr>
 <td class="grey">
@@ -3819,6 +3822,7 @@
 		<div class="asb">
           <a class="b" href="$this_script?file=$conf&amp;menu=small">Dash</a>
           <a class="b" href="$this_script?file=$conf&amp;menu=large">Large Dash</a>
+          <a class="b" href="$opsview_base">Opsview</a>
           <a class="b" href="$NMIS::config{'<url_base>'}/">Doc</a>
           <a class="b" href="$NMIS::config{help_file}">Help</a>
 		</div>
