--- nrpe-2.14/configure.in	2012-12-21 17:22:00.000000000 +0000
+++ nrpe-2.14.fixed/configure.in	2013-02-25 14:03:40.000000000 +0000
@@ -4,8 +4,8 @@
 define([AC_CACHE_LOAD],)
 define([AC_CACHE_SAVE],)
 
-2.14([nrpe],[2.13],[nagios-users@lists.sourceforge.net],[nrpe],[http://www.nagios.org])
-AC_CONFIG_SRCDIR([src/nrpe.c])
+dnl 2.14([nrpe],[2.13],[nagios-users@lists.sourceforge.net],[nrpe],[http://www.nagios.org])
+AC_INIT([src/nrpe.c])
 AC_CONFIG_HEADERS([include/config.h])
 AC_CONFIG_FILES([Makefile
 	subst
