--- nrpe-2.14.original/configure.in	2013-02-25 13:53:46.000000000 +0000
+++ nrpe-2.14/configure.in	2013-02-25 14:03:40.000000000 +0000
@@ -303,9 +303,11 @@
 	        printf "SSL headers found in $ssldir\n";
 
 		dnl Now try and find SSL libraries
+        dnl Opsview patch: Added $ssldir/lib /usr/lib/i386-linux-gnu /usr/lib/x86_64-linux-gnu
+        dnl Latter two for Ubuntu12
 		AC_MSG_CHECKING(for SSL libraries)
 		found_ssl=no
-		for dir in $ssl_lib_dir $ssl_dir /usr/lib64 /usr/lib /usr/local/lib /usr/lib/ssl /usr/ssl/lib /usr/openssl/lib /usr/pkg/lib  /usr/freeware/lib/openssl /usr/sfw/lib /opt/freeware/lib; do
+		for dir in $ssl_lib_dir $ssl_dir /usr/lib64 /usr/lib /usr/local/lib /usr/lib/ssl /usr/ssl/lib /usr/openssl/lib /usr/pkg/lib  /usr/freeware/lib/openssl /usr/sfw/lib /opt/freeware/lib $ssldir/lib /usr/lib/i386-linux-gnu /usr/lib/x86_64-linux-gnu; do
 			ssllibdir="$dir"
 			if test "`uname -s`" == "Darwin" ; then
 				soext="dylib"
