diff -ur scriptaculous-js-1.8.3.original/lib/prototype.js scriptaculous-js-1.8.3/lib/prototype.js
--- scriptaculous-js-1.8.3.original/lib/prototype.js	2009-11-19 10:19:08.000000000 +0000
+++ scriptaculous-js-1.8.3/lib/prototype.js	2009-11-19 10:25:47.000000000 +0000
@@ -2371,7 +2371,7 @@
   },
 
   getOffsetParent: function(element) {
-    if (element.offsetParent) return $(element.offsetParent);
+    if (element.offsetParent && Element.visible(element)) return $(element.offsetParent);
     if (element == document.body) return $(element);
 
     while ((element = element.parentNode) && element != document.body)
